test/e2e/common/node/pod_level_resources.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: *metadata,
+
+		Spec: v1.PodSpec{
+			Containers: testContainers,
+			Volumes: []v1.Volume{
+				{
+					Name: "sysfscgroup",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/cgroup"},
+					},
+				},
+			},
+		},
+	}
+
+	if podResources != nil {
+		res := getResourceInfo(podResources)
+		pod.Spec.Resources = &res
+	}
+
+	return pod
+}
+
+func VerifyPodResources(gotPod v1.Pod, expectedInfo ResourceInfo) {
+	ginkgo.GinkgoHelper()
+	expectedResources := getResourceInfo(&expectedInfo)
+	gomega.Expect(gotPod.Spec.Resources).To(gomega.Equal(&expectedResources))
+}
+
+func VerifyQoS(gotPod v1.Pod, expectedQoS v1.PodQOSClass) {
+	ginkgo.GinkgoHelper()
+	gomega.Expect(gotPod.Status.QOSClass).To(gomega.Equal(expectedQoS))
+}
+
+func VerifyPodCgroups(ctx context.Context, f *framework.Framework, pod *v1.Pod, expectedResources ResourceInfo) error {
+	ginkgo.GinkgoHelper()
+	if pod.Spec.Resources != nil {
+		cmd := fmt.Sprintf("find /tmp -name '*%s*'", strings.ReplaceAll(string(pod.UID), "-", "_"))
+		framework.Logf("Namespace %s Pod %s container - finding Pod cgroup directory path: %q", cmd)
+		podCgPath, stderr, err := e2epod.ExecCommandInContainerWithFullOutput(f, pod.Name, pod.Spec.Containers[0].Name, []string{"/bin/sh", "-c", cmd}...)
+		if err != nil || len(stderr) > 0 {
+			return fmt.Errorf("encountered error while running command: %q, \nerr: %v \nstdErr: %q", cmd, err, stderr)
+		}
+
+		cpuWeightCgPath := fmt.Sprintf("%s/cpu.weight", podCgPath)
+		expectedCPUShares := int64(kubecm.MilliCPUToShares(pod.Spec.Resources.Requests.Cpu().MilliValue()))
+		// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+		// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+		expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
+		err = e2epod.VerifyCgroupValue(f, pod, pod.Spec.Containers[0].Name, cpuWeightCgPath, strconv.FormatInt(expectedCPUShares, 10))
+		if err != nil {
+			return fmt.Errorf("failed to verify cpu request cgroup value: %v", err)
+		}
+
+		cpuLimCgPath := fmt.Sprintf("%s/cpu.max", podCgPath)
+		cpuQuota := kubecm.MilliCPUToQuota(pod.Spec.Resources.Limits.Cpu().MilliValue(), kubecm.QuotaPeriod)
+		expectedCPULimit := strconv.FormatInt(cpuQuota, 10)
+		expectedCPULimit = fmt.Sprintf("%s %s", expectedCPULimit, CPUPeriod)
+		err = e2epod.VerifyCgroupValue(f, pod, pod.Spec.Containers[0].Name, cpuLimCgPath, expectedCPULimit)
+		if err != nil {
+			return err
+		}
+
+		memLimCgPath := fmt.Sprintf("%s/memory.max", podCgPath)
+		expectedMemLim := strconv.FormatInt(pod.Spec.Resources.Limits.Memory().Value(), 10)
+		err = e2epod.VerifyCgroupValue(f, pod, pod.Spec.Containers[0].Name, memLimCgPath, expectedMemLim)
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func podLevelResourcesTests(f *framework.Framework) {
```

Comment:
Could you also add a test to ensure that when the feature is enabled, but pod resources are not set, the container resources and cgroup values are not affected? If this is already covered by other test case, feel free to answer and resolve the comment.